chain/network/src/network_protocol/borsh_conv.rs

```diff
@@ -177,6 +177,9 @@ impl From<&mem::PeerMessage> for net::PeerMessage {
                 net::PeerMessage::SyncRoutingTable(rtu.into())
             }
             mem::PeerMessage::RequestUpdateNonce(e) => net::PeerMessage::RequestUpdateNonce(e),
+            mem::PeerMessage::DistanceVector(_) => {
+                panic!("DistanceVector is not supported in Borsh encoding")
+            }
```

Comment:
Good question, I will investigate it. For now I think it shouldn't block this PR since it is an existing practice for other types: https://github.com/near/nearcore/blob/f50201d62848ee770dde5bab6a084ef35ad54029/chain/network/src/network_protocol/borsh_conv.rs#L169

The new serialization for network messages is protocol buffers (using rust protobuf crate) rather than borsh. The rust serde crate is used on top to serialize/deserialize rust data structures.